keps/sig-storage/3476-volume-group/README.md

```diff
@@ -0,0 +1,1670 @@
+  // entries is more than this number, the Plugin MUST set `next_token`
+  // in the response which can be used to get the next page of entries
+  // in the subsequent `ListVolumeGroupSnapshots` call. This field is OPTIONAL. If
+  // not specified (zero value), it means there is no restriction on the
+  // number of entries that can be returned.
+  // The value of this field MUST NOT be negative.
+  int32 max_entries = 1;
+
+  // A token to specify where to start paginating. Set this field to
+  // `next_token` returned by a previous `ListVolumeGroupSnapshots` call to get the
+  // next page of entries. This field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string starting_token = 2;
+}
+
+message ListVolumeGroupSnapshotsResponse {
+  option (alpha_message) = true;
+
+  message Entry {
+    // This field is REQUIRED
+    VolumeGroupSnapshot group_snapshot = 1;
+  }
+
+  repeated Entry entries = 1;
+
+  // This token allows you to get the next page of entries for
+  // `ListVolumeGroupSnapshots` request. If the number of entries is larger than
+  // `max_entries`, use the `next_token` as a value for the
+  // `starting_token` field in the next `ListVolumeGroupSnapshots` request. This
+  // field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string next_token = 2;
+}
+```
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Other
+    - Describe the mechanism:
+      The external volume group and group snapshot controllers do not have a
+      feature gate because they are out of tree.
+      It is enabled when these external controller sidecars are deployed with the CSI driver.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+      From the controller side, it only affects the external controller sidecars.
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      No.
+
+* **Does enabling the feature change any default behavior?**
+  Yes. Enabling the feature can allow a new PVC to be created and added to a VolumeGroup. Enabling the feature can also allow a VolumeSnapshot to be created as part of the VolumeSnapshotGroup.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes. In order to disable this feature once it has been enabled, we first need to make sure that all VolumeGroup and VolumeGroupSnapshot API objects are deleted. Then the new controllers for VolumeGroup and VolumeGroupSnapshot can be stopped/removed, and external-provisioner sidecar and external-snapshotter controller/sidecar can be downgraded to a version without this feature.
```

Comment:
What happens if we don't delete those but just remove the sidecars? That is, where will we see errors / failures if different parts of the solution are not present (controllers and/or sidecars)?

Reply:
If we don't delete the VolumeGroup and VolumeGroupSnapshot API objects and CRDs but just remove the sidecars, they continue to exist in the API server. User may delete an individual PVC that is part of a VolumeGroup or delete an individual VolumeSnapshot that is associated with a VolumeGroupSnapshot. After that if user starts the sidecars again and try to use the pre-existing VolumeGroup and VolumeGroupSnapshot, they are no longer in sync with the storage system.